docs/source/tech_notes.rst

```diff
@@ -0,0 +1,67 @@
+
+Here is a list of statistics that can be useful:
+
+- `S local`: the tensor :math:`S_{-1}` for the natural gradient
+- `M`: the tensor :math:`M_{-1}` for the natural gradient
+- `S prev`: the tensor :math:`S_{-2}` to compute the new weights
+- `M prev`: the tensor :math:`M_{-2}` to compute the new weights
+- `cross covariance`: the cross covariance :math:`P` to compute the new weights
+
+To compute those statistics we need to access some intermediate quantities:
+
+- :math:`a`: the pre-activation of the layer
+- :math:`\nabla_{a} \mathcal{L}`: the gradient of the loss with respect to the pre-activation
+- :math:`b_{-1}`: the activation of the previous layer
+- :math:`b_{-2}`: the activation of two layer before
+
+We store those quantities in the `GroMo` module directly. Depending of the situation it can be stored in different places but it is transparent for the user as we surcharge `input` and `pre_activity` to access those quantities in the same way.
+
+.. image:: images/gromo_links.png
+    :width: 800px
+    :align: center
+    :height: 565px
+    :alt: Where are quantities stored ?
+
+When we link only `GrowingModule` modules together the quantities are stored in the `GrowingModule` module. When we link `GrowingModule` with `AdditionGrowingModule` the quantities are stored in the `AdditionGrowingModule`.  This is due to the fact that we use the `AdditionGrowingModule` when we connect multiple `GrowingModule` together and we do not want to store the same quantities multiple times. Hence we store them in the `AdditionGrowingModule` and we can access them from the `GrowingModule`. Note that we perform the activation after the addition in the `AdditionGrowingModule` so that we can access the pre-activity of the previous layer as the input of the next layer.
+
+=====================
+Growing a layer
+=====================
+
+Sequential case
+----------------
+
+To grow a layer of neurons we need new incoming connections and new outgoing connections. If we look a at section with two modules `M1` and `M2` with a layer of neurons in between we can grow the layer of neurons by adding new outgoing connections in `M1` (noted `O1`) and new incoming connections in `M2` (noted `I2`). To compute the new weights with the TINY/FoGro method we can use the method `compute_optimal_added_parameters` in `M2` that will also update the module `M1` if the option `update_previous` is set to `True`.
+
+You can then use the `extended_forward` method to compute the forward pass with the new weights. More precisely, the `extended_forward` with integrate the `scaling_factor` (noted :math:`\gamma`) to the forward pass:
+
+- :math:`\texttt{extended\_forward}(x) = (M1(x), \gamma  O1(x))`
+- :math:`\texttt{extended\_forward}(M1(x), \gamma  O1(x)) = M2(M1(x)) + \gamma^2  I2(O1(x))`
```

Comment:
extended_forward is incorrectly rendered with the \

Reply:
Yes, I fixed it !